py/test/selenium/webdriver/common/upload_tests.py

```diff
@@ -15,35 +15,63 @@
 # specific language governing permissions and limitations
 # under the License.
 
+import os
+import pytest
 from selenium.webdriver.common.by import By
+from selenium.webdriver.remote.webelement import WebElement
 
 
-import os
+@pytest.fixture
+def get_local_path():
+    current_dir = os.path.dirname(os.path.realpath(__file__))
+
+    def wrapped(filename):
+        return os.path.join(current_dir, filename)
 
+    return wrapped
 
-def test_can_upload_file(driver, pages):
+
+def test_can_upload_file(driver, pages, get_local_path):
 
     pages.load("upload.html")
-    current_dir = os.path.dirname(os.path.realpath(__file__))
-    driver.find_element(By.ID, 'upload').send_keys(os.path.join(current_dir, "test_file.txt"))
+
+    driver.find_element(By.ID, 'upload').send_keys(get_local_path("test_file.txt"))
     driver.find_element(By.ID, 'go').click()
     driver.switch_to.frame(driver.find_element(By.ID, "upload_target"))
     body = driver.find_element(By.CSS_SELECTOR, "body").text
 
     assert "test_file.txt" in body
 
 
-def test_can_upload_two_files(driver, pages):
+def test_can_upload_two_files(driver, pages, get_local_path):
 
     pages.load("upload.html")
-    current_dir = os.path.dirname(os.path.realpath(__file__))
-    driver.find_element(By.ID, 'upload')\
-        .send_keys(
-            os.path.join(current_dir, "test_file.txt") + "\n" + os.path.join(current_dir, "test_file2.txt")
-    )
+    two_file_paths = get_local_path("test_file.txt") + "\n" + get_local_path("test_file2.txt")
+    driver.find_element(By.ID, 'upload').send_keys(two_file_paths)
     driver.find_element(By.ID, 'go').click()
     driver.switch_to.frame(driver.find_element(By.ID, "upload_target"))
     body = driver.find_element(By.CSS_SELECTOR, "body").text
 
     assert "test_file.txt" in body
     assert "test_file2.txt" in body
+
+
```

Comment:
could you decorate these with the following as they won't work unless it goes down the remote

```suggestion
@pytest.mark.xfail_firefox
@pytest.mark.xfail_chrome
@pytest.mark.xfail_safari
```

Reply:
good point, forgot about non-remote drivers :)
fixed